esp-hal/src/i2c/master/mod.rs

```diff
@@ -918,9 +927,11 @@ fn configure_clock(
         // timeout mechanism
         cfg_if::cfg_if! {
             if #[cfg(esp32)] {
-                register_block
-                    .to()
-                    .write(|w| w.time_out().bits(unwrap!(timeout)));
+                if let Some(timeout_value) = timeout {
```

Comment:
It's impossible for this to be `None` currently, I'd prefer keeping the `unwrap!`.

Reply:
Ok, will be done.
Then new `InvalidTimeout` error should probably be gone too